File: python/benchmark/report.py
import json
import os

RESULTS_FILE = "benchmark_results.jsonl"

def save_result(broker, role, messages, duration):
    result = {
        "broker": broker,
        "role": role,
        "messages": messages,
        "duration": duration,
        "throughput": messages / duration if duration > 0 else 0
    }
    
    with open(RESULTS_FILE, 'a') as f:
        f.write(json.dumps(result) + "\n")

File: typescript/benchmark/report.ts
import * as fs from 'fs';
import * as path from 'path';

const RESULTS_FILE = "benchmark_results.jsonl";

export function saveResult(broker: string, role: string, messages: number, duration: number) {
    const result = {
        broker,
        role,
        messages,
        duration,
        throughput: duration > 0 ? messages / duration : 0
    };
    
    // Append to file
    fs.appendFileSync(RESULTS_FILE, JSON.stringify(result) + "\n");
}
